protocol/contracts/dao/Comptroller.sol

```diff
@@ -57,11 +57,13 @@ contract Comptroller is Setters {
         balanceCheck();
     }
 
-    function increaseDebt(uint256 amount) internal {
+    function increaseDebt(uint256 amount) internal returns (uint256) {
         incrementTotalDebt(amount);
-        resetDebt(Constants.getDebtRatioCap());
+        uint256 lessDebt = resetDebt(Constants.getDebtRatioCap());
 
         balanceCheck();
+
+        return lessDebt > amount ? 0 : amount.sub(lessDebt);
```

Comment:
qq -- when would `lessDebt > amount`? Seems like `amount` should always be greater or equal if debt is below cap initially. And this function should prevent it from ever going over, if I understand correctly.

Reply:
It's very unlikely, but there could be cases where we've just significantly changed the debt ratio cap or the total supply of `ESD` such that a large amount of the debt is trimmed off here.

This is more of a defensive check since a subtraction underflow here would cause a revert bricking the entire system.